Python/compile.c

```diff
@@ -213,6 +213,9 @@ struct instr {
         struct instr *_instr__ptr_ = (I); \
         _instr__ptr_->i_opcode = (OP); \
         _instr__ptr_->i_oparg = (ARG); \
+        _instr__ptr_->i_oparg1 = UNUSED_OPARG; \
+        _instr__ptr_->i_oparg2 = UNUSED_OPARG; \
+        _instr__ptr_->i_oparg3 = UNUSED_OPARG; \
```

Comment:
IIRC this is not the first time we've had an issue caused by looking at the oparg of a NOP (or perhaps another oparg-less instruction). Maybe the code that looks at opargs should first check if the given opcode even _has_ an oparg?

Reply:
For the stack opcodes we have the HAS_ARG macro which is the only way to determine whether there is an oparg or not. The compiler only uses it for sanity checks like oparg == 0 unless HAS_ARG.  It's also used in debug prints and dis.

For register ops we don't have the equivalent. In the compiler we don't need to ({we have the oparg_type indicating which ones are used) but there is no harm in having it to sanity check during assembly that code-gen created instructions with the correct number of args, and we will need something for dis. We could generate this from the bytecode definitions (but we'll need to do something for the peruse-instructions, which HAS_ARG supports).

Reply:
I was thinking of issues in the compiler, but actually it may have been the assembler.

Anyways, generating tables from bytecodes.c is a long-term goal, maybe I could accelerate that.